File: AQ_LIB/resources/api/RExamples/CurveMarketDataDisplay/M_CurveMarketDataDisplay.R
dyn.load("MLIB_CLIENT_API")
source("MLIB_CLIENT_API.R")
getLoadedDLLs()
setUpMLIB("","")
meUtilityVersion()


filename <-c("E:/SavedCurves/OISCURVEMARKETDATA.TXT")
oisdata <- meLWOLoad(filename)


# Read each block of data into a variable

propertiesName <- c( "MARKETDATAPROPERTIES" )
properties <- meLWOCurveMarketDataDisplay( oisdata, propertiesName)

oisName <- c("OIS")
oisSwaps <- meLWOCurveMarketDataDisplay( oisdata, oisName)

liborOisBasisName <- c("LIBOROISBASISSPREADS")
liborOisBasisSwaps <- meLWOCurveMarketDataDisplay( oisdata, liborOisBasisName)

liborSwapsName <- c("SWAPS")
liborSwaps <- meLWOCurveMarketDataDisplay( oisdata, liborSwapsName)

# Display one of the blocks
liborSwaps

File: AQ_LIB/resources/api/PythonExamples/CurveMarketDataDisplay/M_CurveMarketDataDisplay.py
#Import the MLIB CLIENT API
import MLIB_CLIENT_API as m

#Load Calendars and Config Files
m.setUpMLIB("","")

#Get the MLIB Version
m.meUtilityVersion()

# Load a saved market data file

filename="E:\SavedCurves\OISCURVEMARKETDATA.TXT"
oisdata =m.meLWOLoad(filename)

# Read each block of data into a variable

propertiesName = "MARKETDATAPROPERTIES"
properties = m.meLWOCurveMarketDataDisplay( oisdata, propertiesName)

oisName = "OIS"
oisSwaps = m.meLWOCurveMarketDataDisplay( oisdata, oisName)

liborOisBasisName = "LIBOROISBASISSPREADS"
liborOisBasisSwaps = m.meLWOCurveMarketDataDisplay( oisdata, liborOisBasisName)

liborSwapsName = "SWAPS"
liborSwaps = m.meLWOCurveMarketDataDisplay( oisdata, liborSwapsName)

# Display each block of data as a matrix

for row in properties:
  for item in row:
    print item,
  print
  
for row in oisSwaps:
  for item in row:
    print item,
  print
  
for row in liborOisBasisSwaps:
  for item in row:
    print item,
  print

for row in liborSwaps:
  for item in row:
    print item,
  print
  
